### PR TITLE
add basic token Auth

### LINKED ISSUE #2: New feature: add Basic Auth
I'm working on a bit of software that uses `weblog` against an AWS lambda. For security reasons, I need to enable any type of authentication.

I've noticed that  Nextflow `SimpleHttpClient` works ([link](https://github.com/nextflow-io/nextflow/blob/8a4348939949ce5532179cc8067350c9c2c598ef/modules/nextflow/src/main/groovy/nextflow/util/SimpleHttpClient.groovy#L232)) that is used by `nf-weblog` already supports Basic authentication and update this plugin with a new bit of configuration to use `httpClient.setBasicToken`.

I've forked the code and added some code changes that should allow this new bit of functionality
https://github.com/nextflow-io/nf-weblog/compare/master...redbass:nf-weblog:add-basictoken-auth?expand=1

Please let me know what is the best way to discuss this solution, my company really needs this bit of functionality to complete our product.

### PR DESCRIPTION
**Pull request by redbass**
This functionality allows to specify `basicToken` as part of `weblog` configuration. The value will be used added to the `Authorization` header when an HTTP request is processed.

### CODE REVIEW

**Comment by bentsherman** [comment]
Can you test the plugin with your private endpoint?

**Comment by redbass** [comment]
I will do a test on Monday.

**Comment by redbass** [comment]
As specified by the instructions in the section _Testing and debugging_ I've run this command:
```
./launch.sh run nextflow-io/hello -plugins nf-weblog -c config.config`
```
where `config.config` has this content:
```
weblog.enabled = 1
weblog.url = 'http://localhost:5000/protected'
weblog.basicToken = '12312312312'
```
that generates this request:
```
Content-Type: application/json
User-Agent: Nextflow/23.08.1-edge
Authorization: Basic MTIzMTIzMTIzMTI=
Host: localhost:5000
Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2
Connection: keep-alive
Content-Length: 24946
```

This code is against a local `Flask` app.

**Review by bentsherman — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -59,6 +59,10 @@ If `true` it will send HTTP POST requests to a given url.
 
 The url where to send HTTP POST requests (default: `http://localhost`).
 
+`weblog.basicToken`
+
+A Basic authentication token to append to each HTTP request.
```
Comment (pditommaso) [inline README.md:64]: This is too generic. If I'm understanding correctly the user should specify the password as a base64 encoded string. This should be explained in the readme.

Comment (bentsherman) [reply README.md:64]: done

**Comment by bentsherman** [comment]
@redbass do you think it would be better to just put the username and password in the configuration, instead of the basic token? That way Nextflow could compute the basic token for you.

**Comment by redbass** [comment]
It would be nice!

Would also be nice to support API Key that is the default authentication method for AWS gateway

This needs a small change in NF http client, but would make life easier.

**Comment by bentsherman** [comment]
okay I will make an issue for now

**Pull request merged**